### PR TITLE
Activate conda environment into container in Dockerfile-celery

### LINKED ISSUE #217: docker celary
# Objetivos / Objectivos / Purpose

## Geral / General

Atualizar o dockerfile-celary para usar miniconda ao invés de usar o python do sistema. 

## Específicos / Specific

- [x] Atualizar o dockerfile-celary para usar o miniconda

# Justificativa / Justification

O uso do miniconda ajuda bastante, incluso em relação a instalação de bibliotecas de sistemas.
E os benefícios se potencializam principalmente devido ao conda-forge,  onde existe muito trabalho e dedicação às bibliotecas relacionadas a GIS.

O PIP compila as bibliotecas localmente, aumentando o tempo de instalação das bibliotecas. Instalar biblioteca pré-compiladas pode diminuir bastante o tempo consumido pela instalação.

O erro a seguir ocorre na compilação do rasterio:

```sh
# command
docker-compose up celery-worker 
# error:
Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-vxyvsbfi/rasterio/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-xr0xe75a --python-tag cp37:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/compat.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/crs.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/profiles.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/sample.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/env.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/fill.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/plot.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/errors.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/merge.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/windows.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/warp.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/enums.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/features.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/__init__.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/coords.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/vfs.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/mask.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/transform.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/tool.py -> build/lib.linux-x86_64-3.7/rasterio
  copying ./rasterio/dtypes.py -> build/lib.linux-x86_64-3.7/rasterio
  creating build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/edit_info.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/sample.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/env.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/overview.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/bounds.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/merge.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/warp.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/clip.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/shapes.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/calc.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/insp.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/__init__.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/main.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/info.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/options.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/stack.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/mask.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/helpers.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/transform.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/convert.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  copying ./rasterio/rio/rasterize.py -> build/lib.linux-x86_64-3.7/rasterio/rio
  creating build/lib.linux-x86_64-3.7/rasterio/tools
  copying ./rasterio/tools/merge.py -> build/lib.linux-x86_64-3.7/rasterio/tools
  copying ./rasterio/tools/__init__.py -> build/lib.linux-x86_64-3.7/rasterio/tools
  copying ./rasterio/tools/mask.py -> build/lib.linux-x86_64-3.7/rasterio/tools
  running egg_info
  writing rasterio.egg-info/PKG-INFO
  writing dependency_links to rasterio.egg-info/dependency_links.txt
  writing entry points to rasterio.egg-info/entry_points.txt
  writing requirements to rasterio.egg-info/requires.txt
  writing top-level names to rasterio.egg-info/top_level.txt
  reading manifest file 'rasterio.egg-info/SOURCES.txt'
  writing manifest file 'rasterio.egg-info/SOURCES.txt'
  running build_ext
  building 'rasterio._base' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/rasterio
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/gdal -I/usr/include/python3.7m -c rasterio/_base.c -o build/temp.linux-x86_64-3.7/rasterio/_base.o -Wno-unused-parameter -Wno-unused-function
  rasterio/_base.c: In function ‘__Pyx__ExceptionSave’:
  rasterio/_base.c:23678:21: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
       *type = tstate->exc_type;
                       ^~~~~~~~
                       curexc_type
  rasterio/_base.c:23679:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
       *value = tstate->exc_value;
                        ^~~~~~~~~
                        curexc_value
  rasterio/_base.c:23680:19: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
       *tb = tstate->exc_traceback;
                     ^~~~~~~~~~~~~
                     curexc_traceback
  rasterio/_base.c: In function ‘__Pyx__ExceptionReset’:
  rasterio/_base.c:23687:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
       tmp_type = tstate->exc_type;
                          ^~~~~~~~
                          curexc_type
  rasterio/_base.c:23688:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
       tmp_value = tstate->exc_value;
                           ^~~~~~~~~
                           curexc_value
  rasterio/_base.c:23689:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
       tmp_tb = tstate->exc_traceback;
                        ^~~~~~~~~~~~~
                        curexc_traceback
  rasterio/_base.c:23690:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
       tstate->exc_type = type;
               ^~~~~~~~
               curexc_type
  rasterio/_base.c:23691:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
       tstate->exc_value = value;
               ^~~~~~~~~
               curexc_value
  rasterio/_base.c:23692:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
       tstate->exc_traceback = tb;
               ^~~~~~~~~~~~~
               curexc_traceback
  rasterio/_base.c: In function ‘__Pyx__GetException’:
  rasterio/_base.c:23737:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
       tmp_type = tstate->exc_type;
                          ^~~~~~~~
                          curexc_type
  rasterio/_base.c:23738:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
       tmp_value = tstate->exc_value;
                           ^~~~~~~~~
                           curexc_value
  rasterio/_base.c:23739:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
       tmp_tb = tstate->exc_traceback;
                        ^~~~~~~~~~~~~
                        curexc_traceback
  rasterio/_base.c:23740:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
       tstate->exc_type = local_type;
               ^~~~~~~~
               curexc_type
  rasterio/_base.c:23741:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
       tstate->exc_value = local_value;
               ^~~~~~~~~
               curexc_value
  rasterio/_base.c:23742:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
       tstate->exc_traceback = local_tb;
               ^~~~~~~~~~~~~
               curexc_traceback
  rasterio/_base.c: In function ‘__Pyx__ExceptionSwap’:
  rasterio/_base.c:24889:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
       tmp_type = tstate->exc_type;
                          ^~~~~~~~
                          curexc_type
  rasterio/_base.c:24890:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
       tmp_value = tstate->exc_value;
                           ^~~~~~~~~
                           curexc_value
  rasterio/_base.c:24891:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
       tmp_tb = tstate->exc_traceback;
                        ^~~~~~~~~~~~~
                        curexc_traceback
  rasterio/_base.c:24892:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
       tstate->exc_type = *type;
               ^~~~~~~~
               curexc_type
  rasterio/_base.c:24893:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
       tstate->exc_value = *value;
               ^~~~~~~~~
               curexc_value
  rasterio/_base.c:24894:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
       tstate->exc_traceback = *tb;
               ^~~~~~~~~~~~~
               curexc_traceback
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Running setup.py clean for rasterio
  Running setup.py bdist_wheel for cufflinks: started
  Running setup.py bdist_wheel for cufflinks: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/ef/6a/a4/ff5ceff51314afb5bb24745bf74b51ed4cee53609b66bafa1c
  Running setup.py bdist_wheel for billiard: started
  Running setup.py bdist_wheel for billiard: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/f0/69/3c/7df7f2f49f3934716458ea8a3ca5f3263ce2656e5ed5322acd
  Running setup.py bdist_wheel for static3: started
  Running setup.py bdist_wheel for static3: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/c3/d0/4b/a415104b0303aa18d566463b02d1eacb8f43bc3d29a6dc4574
  Running setup.py bdist_wheel for retrying: started
  Running setup.py bdist_wheel for retrying: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/d7/a9/33/acc7b709e2a35caa7d4cae442f6fe6fbf2c43f80823d46460c
  Running setup.py bdist_wheel for backcall: started
  Running setup.py bdist_wheel for backcall: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/98/b0/dd/29e28ff615af3dda4c67cab719dd51357597eabff926976b45
  Running setup.py bdist_wheel for pyrsistent: started
  Running setup.py bdist_wheel for pyrsistent: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/83/59/9a/a037b9b3c3e93d9275ea0aff9d6064400f372879dfdab01afe
  Running setup.py bdist_wheel for tornado: started
  Running setup.py bdist_wheel for tornado: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/61/7e/7a/5e02e60dc329aef32ecf70e0425319ee7e2198c3a7cf98b4a2
  Running setup.py bdist_wheel for prometheus-client: started
  Running setup.py bdist_wheel for prometheus-client: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/4b/04/b8/3709c73e7453f311ebd46ad581b89642543213f995e2659b9e
  Running setup.py bdist_wheel for pandocfilters: started
  Running setup.py bdist_wheel for pandocfilters: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/39/01/56/f1b08a6275acc59e846fa4c1e1b65dbc1919f20157d9e66c20
Successfully built django-bootstrap4 django-leaflet python-decouple sqlalchemy django-chunked-upload dj-static cufflinks billiard static3 retrying backcall pyrsistent tornado prometheus-client pandocfilters
Failed to build rasterio
Installing collected packages: django, django-bootstrap4, django-leaflet, gunicorn, python-decouple, sqlalchemy, python-memcached, django-makemessages-xgettext, pbr, mock, dbfread, dj-database-url, django-chunked-upload, vine, amqp, kombu, billiard, celery, static3, dj-static, affine, snuggs, rasterio, sh, urllib3, idna, certifi, requests, retrying, pyrsistent, jsonschema, ipython-genutils, traitlets, jupyter-core, nbformat, plotly, colorlover, parso, jedi, pygments, wcwidth, prompt-toolkit, backcall, ptyprocess, pexpect, pickleshare, ipython, pyzmq, tornado, jupyter-client, ipykernel, terminado, MarkupSafe, jinja2, prometheus-client, Send2Trash, testpath, defusedxml, bleach, mistune, pandocfilters, nbconvert, notebook, widgetsnbextension, ipywidgets, cufflinks
  Running setup.py install for rasterio: started
    Running setup.py install for rasterio: finished with status 'error'
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-vxyvsbfi/rasterio/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-022fnplu/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/compat.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/crs.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/profiles.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/sample.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/env.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/fill.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/plot.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/errors.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/merge.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/windows.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/warp.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/enums.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/features.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/__init__.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/coords.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/vfs.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/mask.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/transform.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/tool.py -> build/lib.linux-x86_64-3.7/rasterio
    copying ./rasterio/dtypes.py -> build/lib.linux-x86_64-3.7/rasterio
    creating build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/edit_info.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/sample.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/env.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/overview.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/bounds.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/merge.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/warp.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/clip.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/shapes.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/calc.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/insp.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/__init__.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/main.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/info.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/options.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/stack.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/mask.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/helpers.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/transform.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/convert.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    copying ./rasterio/rio/rasterize.py -> build/lib.linux-x86_64-3.7/rasterio/rio
    creating build/lib.linux-x86_64-3.7/rasterio/tools
    copying ./rasterio/tools/merge.py -> build/lib.linux-x86_64-3.7/rasterio/tools
    copying ./rasterio/tools/__init__.py -> build/lib.linux-x86_64-3.7/rasterio/tools
    copying ./rasterio/tools/mask.py -> build/lib.linux-x86_64-3.7/rasterio/tools
    running egg_info
    writing rasterio.egg-info/PKG-INFO
    writing dependency_links to rasterio.egg-info/dependency_links.txt
    writing entry points to rasterio.egg-info/entry_points.txt
    writing requirements to rasterio.egg-info/requires.txt
    writing top-level names to rasterio.egg-info/top_level.txt
    reading manifest file 'rasterio.egg-info/SOURCES.txt'
    writing manifest file 'rasterio.egg-info/SOURCES.txt'
    running build_ext
    building 'rasterio._base' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/rasterio
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/gdal -I/usr/include/python3.7m -c rasterio/_base.c -o build/temp.linux-x86_64-3.7/rasterio/_base.o -Wno-unused-parameter -Wno-unused-function
    rasterio/_base.c: In function ‘__Pyx__ExceptionSave’:
    rasterio/_base.c:23678:21: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         *type = tstate->exc_type;
                         ^~~~~~~~
                         curexc_type
    rasterio/_base.c:23679:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         *value = tstate->exc_value;
                          ^~~~~~~~~
                          curexc_value
    rasterio/_base.c:23680:19: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         *tb = tstate->exc_traceback;
                       ^~~~~~~~~~~~~
                       curexc_traceback
    rasterio/_base.c: In function ‘__Pyx__ExceptionReset’:
    rasterio/_base.c:23687:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tmp_type = tstate->exc_type;
                            ^~~~~~~~
                            curexc_type
    rasterio/_base.c:23688:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tmp_value = tstate->exc_value;
                             ^~~~~~~~~
                             curexc_value
    rasterio/_base.c:23689:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tmp_tb = tstate->exc_traceback;
                          ^~~~~~~~~~~~~
                          curexc_traceback
    rasterio/_base.c:23690:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tstate->exc_type = type;
                 ^~~~~~~~
                 curexc_type
    rasterio/_base.c:23691:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tstate->exc_value = value;
                 ^~~~~~~~~
                 curexc_value
    rasterio/_base.c:23692:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tstate->exc_traceback = tb;
                 ^~~~~~~~~~~~~
                 curexc_traceback
    rasterio/_base.c: In function ‘__Pyx__GetException’:
    rasterio/_base.c:23737:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tmp_type = tstate->exc_type;
                            ^~~~~~~~
                            curexc_type
    rasterio/_base.c:23738:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tmp_value = tstate->exc_value;
                             ^~~~~~~~~
                             curexc_value
    rasterio/_base.c:23739:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tmp_tb = tstate->exc_traceback;
                          ^~~~~~~~~~~~~
                          curexc_traceback
    rasterio/_base.c:23740:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tstate->exc_type = local_type;
                 ^~~~~~~~
                 curexc_type
    rasterio/_base.c:23741:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tstate->exc_value = local_value;
                 ^~~~~~~~~
                 curexc_value
    rasterio/_base.c:23742:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tstate->exc_traceback = local_tb;
                 ^~~~~~~~~~~~~
                 curexc_traceback
    rasterio/_base.c: In function ‘__Pyx__ExceptionSwap’:
    rasterio/_base.c:24889:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tmp_type = tstate->exc_type;
                            ^~~~~~~~
                            curexc_type
    rasterio/_base.c:24890:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tmp_value = tstate->exc_value;
                             ^~~~~~~~~
                             curexc_value
    rasterio/_base.c:24891:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tmp_tb = tstate->exc_traceback;
                          ^~~~~~~~~~~~~
                          curexc_traceback
    rasterio/_base.c:24892:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tstate->exc_type = *type;
                 ^~~~~~~~
                 curexc_type
    rasterio/_base.c:24893:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tstate->exc_value = *value;
                 ^~~~~~~~~
                 curexc_value
    rasterio/_base.c:24894:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tstate->exc_traceback = *tb;
                 ^~~~~~~~~~~~~
                 curexc_traceback
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-vxyvsbfi/rasterio/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-022fnplu/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-vxyvsbfi/rasterio/
ERROR: Service 'celery-worker' failed to build: The command '/bin/sh -c pip3 install -r /tmp/requirements.txt' returned a non-zero code: 1
```

### LINKED ISSUE #325: Configure conda in Dockerfile-celery
# Objetivos / Objectivos / Purpose
- Resolver conflitos de build
- Padronizar containers

### PR DESCRIPTION
**Pull request by esloch**
- Closes #217
- Closes #325

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -6,35 +6,35 @@ RUN apt-get -qq update --yes \
...
+    && update-locale LANG=pt_BR.UTF-8
 
 # Create deploy user
 RUN useradd --shell=/bin/bash --home=/srv/deploy/ --create-home deploy
 
+# Configure conda-channels and install mamba
+RUN conda config --add channels conda-forge \
+  && conda update --all --yes --quiet \
+  && conda install --yes conda-build mamba \
+  && conda clean --all --yes
```
Comment (xmnlab) [inline docker/Dockerfile:21]: maybe you can change it to `&& conda clean -afy`

File: docker/Dockerfile [hunk 2]
```diff
@@ -6,35 +6,35 @@ RUN apt-get -qq update --yes \
...
-RUN /opt/conda/bin/conda config --add channels conda-forge \
-  && /opt/conda/bin/conda update --all --yes --quiet \
-  && /opt/conda/bin/conda env update --name base --file /tmp/environment.yml \
-  && /opt/conda/bin/conda install --yes conda-build \
-  && /opt/conda/bin/conda clean --all --yes
-
-ENV PATH="/opt/conda/bin:${PATH}"
+# Use environment to update the env base
+RUN mamba env update --file /tmp/environment.yml --name base \
+  && mamba clean --all --yes
```
Comment (xmnlab) [inline docker/Dockerfile:29]: I think mamba here doesn't improve anything

maybe you can change it to `&& conda clean -afy`

File: docker/Dockerfile [hunk 3]
```diff
@@ -6,35 +6,35 @@ RUN apt-get -qq update --yes \
...
+# Copy AlertaDengue into home directory and fix permissions
 RUN mkdir /srv/deploy/logs
 COPY --chown=deploy AlertaDengue  /srv/deploy/AlertaDengue
 
 WORKDIR /srv/deploy/AlertaDengue
 
 EXPOSE 8000
 USER deploy
-CMD ["/srv/deploy/AlertaDengue/runwsgi.sh"]
+CMD [“conda”, “env”, “create”, “source activate base"]
```
Comment (xmnlab) [inline docker/Dockerfile:39]: does it really need this?

File: docker/Dockerfile-celery [hunk 4]
```diff
@@ -1,46 +1,53 @@
...
-# helps build the container a *lot* faster by using the cache.
-ADD requirements.txt /tmp/requirements.txt
+# Copy environment file to tmp/
+ARG PYTHON_VERSION
+COPY environment-${PYTHON_VERSION}.yml /tmp/environment.yml
 
-RUN pip3 install -r /tmp/requirements.txt
+# Use environment to update the env base
+RUN mamba env update --file /tmp/environment.yml --name base \
+  && mamba clean --all --yes
```
Comment (xmnlab) [inline docker/Dockerfile-celery:29]: maybe you can change it to `&& conda clean -afy`

File: docker/Dockerfile-celery [hunk 5]
```diff
@@ -1,46 +1,53 @@
...
-# Change the permissions for the user home directory
-RUN chown -R deploy:deploy /srv/deploy/
-
-RUN pip3 install celery
+RUN cp -p /srv/deploy/example_env_file /srv/deploy/.env_staging
 
 USER deploy
 
-CMD ["/srv/deploy/AlertaDengue/run_celery_worker.sh"]
+CMD [“conda”, “env”, “create”, “source activate base"]
```
Comment (xmnlab) [inline docker/Dockerfile-celery:52]: does it really this?

File: docker/prepare_permission.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Basic if statement
+if [[ ${HOST_UID} != 0 ]]
```
Comment (xmnlab) [inline docker/prepare_permission.sh:4]: I think here it should be != ""

File: docker/Dockerfile (hunk 1 above)
Comment (esloch) [reply docker/Dockerfile:21]: ok.
Thanks @xmnlab!

File: docker/Dockerfile (hunk 3 above)
Comment (esloch) [reply docker/Dockerfile:39]: It is not necessary but only ensures that the container goes up with the conda base activated...

File: docker/Dockerfile-celery (hunk 5 above)
Comment (esloch) [reply docker/Dockerfile-celery:52]: the run_celery_worker.sh is now run at entrypoint.

File: docker/Dockerfile (hunk 3 above)
Comment (esloch) [reply docker/Dockerfile:39]: In our case, it doesn't make sense even to have the entrypoint or even the conda env activated to run the image, i fix this in another commit. 
Thanks

**Pull request merged**